File: core/src/main/kotlin/com/example/toyracers/track/SurfaceType.kt
package com.example.toyracers.track

/** Surface categories consumed by rendering now and driving physics later. */
enum class SurfaceType {
    ASPHALT,
    PARQUET,
    TILE,
    GRASS,
    BOOST,
    OIL,
    ;

    val isRoad: Boolean
        get() = this == ASPHALT || this == BOOST || this == OIL
}

File: dart/lib/simulation/surface/surface_type.dart
/// The simulation surfaces defined by the shared compatibility schema.
enum SurfaceType { asphalt, parquet, tile, grass, boost, oil }

/// Reference-defined driving semantics for each [SurfaceType].
///
/// This mapping is part of the compatibility contract. It intentionally does
/// not derive road status from a surface name, track geometry, or styling.
extension SurfaceTypeRoadStatus on SurfaceType {
  bool get isRoad => switch (this) {
    SurfaceType.asphalt || SurfaceType.boost || SurfaceType.oil => true,
    SurfaceType.parquet || SurfaceType.tile || SurfaceType.grass => false,
  };
}
